modules/nlu/src/backend/tools/fastText.ts

```diff
@@ -0,0 +1,235 @@
+import { ChildProcess, execFile, ExecFileOptions, execFileSync } from 'child_process'
+import fs from 'fs'
+import { EOL } from 'os'
+import { join } from 'path'
+
+let bin = 'ft_linux'
+if (process.platform === 'win32') {
+  bin = 'ft_win.exe'
+} else if (process.platform === 'darwin') {
+  bin = 'ft_osx'
+}
+
+export type FastTextTrainArgs = {
+  method: 'supervised' | 'skipgram' | 'cbow'
+  learningRate: number
+  epoch: number
+  bucket: number
+  dim: number
+  wordGram: number
+  minCount: number
+  minn: number
+  maxn: number
+}
+
+export const DefaultFastTextTrainArgs: FastTextTrainArgs = {
```

Comment:
we'll have to grid search on this for production training or if it's fast enough simply grid search at every training